### PR TITLE
*: update rustc, libraries and fix clippy warnings

### PR DESCRIPTION
**Pull request by disksing**
bump rustc to `1.9.0-nightly (5ab11d72c 2016-04-02)`

### CODE REVIEW

File: src/raft/storage.rs [hunk 1]
```diff
@@ -55,6 +55,12 @@ pub struct MemStorageCore {
...
 }
 
+impl Default for MemStorageCore {
+    fn default() -> MemStorageCore {
+        MemStorageCore::new()
+    }
+}
+
 impl MemStorageCore {
     pub fn new() -> MemStorageCore {
```
Comment (BusyJay) [inline src/raft/storage.rs]: I think this method can be removed.

File: src/raftstore/coprocessor/split_observer.rs [hunk 2]
```diff
@@ -170,52 +170,52 @@ mod test {
         let path = TempDir::new("test-raftstore").unwrap();
         let storage = new_peer_storage(&path);
         let mut ctx = ObserverContext::new(&storage);
-        let mut req = AdminRequest::new();
+        let mut request = AdminRequest::new();
 
         let mut observer = SplitObserver;
 
-        let res = observer.pre_admin(&mut ctx, &mut req);
+        let response = observer.pre_admin(&mut ctx, &mut request);
```
Comment (ngaut) [inline src/raftstore/coprocessor/split_observer.rs]: s/response/resp

File: tests/raftstore/test_compact_log.rs [hunk 3]
```diff
@@ -40,11 +40,11 @@ fn test_compact_log<T: Simulator>(cluster: &mut Cluster<T>) {
 
     for i in 1..1000 {
         let (k, v) = (format!("key{}", i), format!("value{}", i));
-        let putk = k.as_bytes();
-        let putv = v.as_bytes();
-        cluster.put(putk, putv);
-        let v = cluster.get(putk);
-        assert_eq!(v, Some(putv.to_vec()));
+        let put_key = k.as_bytes();
```
Comment (ngaut) [inline tests/raftstore/test_compact_log.rs]: s/put_key/key

File: tests/raftstore/test_compact_log.rs [hunk 4]
```diff
@@ -40,11 +40,11 @@ fn test_compact_log<T: Simulator>(cluster: &mut Cluster<T>) {
...
+        let put_value = v.as_bytes();
```
Comment (ngaut) [inline tests/raftstore/test_compact_log.rs]: s/put_value/value

File: tests/raftstore/test_single.rs [hunk 5]
```diff
@@ -47,18 +47,18 @@ fn test_delete<T: Simulator>(cluster: &mut Cluster<T>) {
 
     for i in 1..1000 {
         let (k, v) = (format!("key{}", i), format!("value{}", i));
-        let putk = k.as_bytes();
-        let putv = v.as_bytes();
-        cluster.put(putk, putv);
-        let v = cluster.get(putk);
-        assert_eq!(v, Some(putv.to_vec()));
+        let put_key = k.as_bytes();
```
Comment (ngaut) [inline tests/raftstore/test_single.rs]: Ditto.

File: tests/raftstore/test_single.rs [hunk 6]
```diff
@@ -68,32 +68,32 @@ fn test_seek<T: Simulator>(cluster: &mut Cluster<T>) {
...
+        let put_value = v.as_bytes();
+        cluster.put(put_key, put_value);
     }
 
     for i in 0..100 {
         let k = format!("key{:03}", i);
-        let putk = k.as_bytes();
-        let (k, v) = cluster.seek(putk).unwrap();
+        let put_key = k.as_bytes();
```
Comment (ngaut) [inline tests/raftstore/test_single.rs]: Ditto.

File: tests/raftstore/test_single.rs [hunk 7]
```diff
@@ -68,32 +68,32 @@ fn test_seek<T: Simulator>(cluster: &mut Cluster<T>) {
 
     for i in 100..200 {
         let (k, v) = (format!("key{}", i), format!("value{}", i));
-        let putk = k.as_bytes();
-        let putv = v.as_bytes();
-        cluster.put(putk, putv);
+        let put_key = k.as_bytes();
```
Comment (ngaut) [inline tests/raftstore/test_single.rs]: Ditto.

File: tests/raftstore/test_single.rs [hunk 8]
```diff
@@ -68,32 +68,32 @@ fn test_seek<T: Simulator>(cluster: &mut Cluster<T>) {
...
     }
 
     for i in 100..200 {
         let (k, v) = (format!("key{}", i), format!("value{}", i));
-        let putk = k.as_bytes();
-        let putv = v.as_bytes();
-        let (sk, sv) = cluster.seek(putk).unwrap();
-        assert_eq!(sk, putk);
-        assert_eq!(sv, putv);
+        let put_key = k.as_bytes();
```
Comment (ngaut) [inline tests/raftstore/test_single.rs]: Ditto.

**Comment by ngaut** [comment]
LGTM

**Comment by disksing** [comment]
@BusyJay PTAL

File: src/raftstore/coprocessor/split_observer.rs [hunk 9]
```diff
@@ -170,52 +170,52 @@ mod test {
         let path = TempDir::new("test-raftstore").unwrap();
         let storage = new_peer_storage(&path);
         let mut ctx = ObserverContext::new(&storage);
-        let mut req = AdminRequest::new();
+        let mut request = AdminRequest::new();
```
Comment (BusyJay) [inline src/raftstore/coprocessor/split_observer.rs]: s/request/req/

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**